aepsych/benchmark/problem.py

```diff
@@ -316,22 +316,26 @@ def evaluate(self, strat: Union[Strategy, SequentialStrategy]) -> Dict[str, floa
 class LSEProblemWithEdgeLogging(LSEProblem):
     eps = 0.05
 
-    def __init__(self, thresholds):
+    def __init__(self, thresholds: Union[float, List, torch.Tensor]) -> None:
         super().__init__(thresholds)
 
-    def evaluate(self, strat):
+    def evaluate(self, strat: Union[SequentialStrategy, Strategy]) -> Dict[str, float]:
         metrics = super().evaluate(strat)
 
         # add number of edge samples to the log
 
         # get the trials selected by the final strat only
-        n_opt_trials = strat.strat_list[-1].n_trials
+        if isinstance(strat, SequentialStrategy):
+            n_opt_trials = strat.strat_list[-1].n_trials
+        elif isinstance(strat, Strategy):
+            n_opt_trials = strat.n_trials
 
         lb, ub = strat.lb, strat.ub
         r = ub - lb
         lb2 = lb + self.eps * r
         ub2 = ub - self.eps * r
 
+        assert isinstance(strat.x, torch.Tensor), "strat.x must be a tensor"
```

Comment:
Shouldn't strat.x always be a tensor now that we don't accept numpy arrays anymore?

Reply:
That's correct. However, `start.x` is initialized as `Optional[torch.Tensor]` since it starts as `None`. Therefore, this check is essential for mypy.